toolchain/driver/driver_main.cpp

```diff
@@ -4,16 +4,26 @@
 
 #include <cstdlib>
 
-#include "llvm/ADT/Sequence.h"
 #include "llvm/ADT/SmallVector.h"
 #include "llvm/ADT/StringRef.h"
+#include "llvm/Support/FileSystem.h"
 #include "toolchain/driver/driver.h"
 
 auto main(int argc, char** argv) -> int {
   if (argc < 1) {
     return EXIT_FAILURE;
   }
 
+  // Behave as if the working directory is where `bazel run` was invoked.
+  char* build_working_dir = getenv("BUILD_WORKING_DIRECTORY");
```

Comment:
It seems problematic to me for our production binary to behave differently if this bazel-specific environment variable is set. For example, because environment variables get inherited, if some Carbon developer writes an `sh_binary` and runs it with `bazel run`, and that shell script internally changes the working directory and then invokes the carbon driver, it will behave in surprising ways.

I think our goal should be that the toolchain doesn't need to be run from a particular working directory in order to find its own files (we shouldn't need the `cd $TOP/bazel-runfiles` that `bazel run` performs); is there any way we can tell `bazel run` to just run the binary and not change directory, perhaps in the BUILD file? If it's important to support `bazel run`, maybe we could add an `sh_binary` that one could `bazel run`, that would `cd $BUILD_WORKING_DIR` and then run the toolchain with the original set of arguments?

Reply:
Hm. Given the existence of `installers/local/carbon.cpp`, is the intent that this is *not* a production binary? That is, that people will only ever be installing and using the busy-boxed binary that lives there? If so, so that this binary is only ever intended to be run out of the bazel build directory, then I retract my objection :)

Reply:
(From in-person discussion:) Yes, that's the intent.